cmd/oras/internal/display/status/interface.go

```diff
@@ -61,3 +63,24 @@ type CopyHandler interface {
 	PostCopy(ctx context.Context, desc ocispec.Descriptor) error
 	OnMounted(ctx context.Context, desc ocispec.Descriptor) error
 }
+
+// ManifestIndexCreateHandler handles status output for manifest index create command.
+type ManifestIndexCreateHandler interface {
+	OnSourceManifestFetching(source string) error
+	OnSourceManifestFetched(source string) error
+	OnIndexPacked(shortDigest string) error
+	OnIndexPushed(path string) error
+}
+
+// ManifestIndexUpdateHandler handles status output for manifest index update command.
+type ManifestIndexUpdateHandler interface {
```

Comment:
Contains duplicated methods `ManifestIndexCreateHandler` as below

```golang
	OnIndexFetching(indexRef string) error
	OnIndexFetched(indexRef string, digest digest.Digest) error
```

You can create a new interface and use it in both interfaces.

Reply:
Created an interface `ManifestReferenceFetchHandler` and both index create and index update handlers contain it.

Reply:
Based on this [comment](https://github.com/oras-project/oras/pull/1509#discussion_r1796515589), this interface is now removed.